plugin.go

```diff
@@ -29,46 +30,56 @@ func (p *Plugin) Exec() error {
 	client := drone.NewClientToken(p.Server, p.Token)
 
 	for _, entry := range p.Repos {
-
-		// parses the repository name in owner/name@branch format
-		owner, name, branch := parseRepoBranch(entry)
+	        owner, name, branch := parseRepoBranch(entry)
 		if len(owner) == 0 || len(name) == 0 {
 			return fmt.Errorf("Error: unable to parse repository name %s.\n", entry)
 		}
-		if p.Fork {
-			// get the latest build for the specified repository
-			build, err := client.BuildLast(owner, name, branch)
-			if err != nil {
-				return fmt.Errorf("Error: unable to get latest build for %s.\n", entry)
-			}
-			// start a new  build
-			_, err = client.BuildFork(owner, name, build.Number)
-			if err != nil {
-				return fmt.Errorf("Error: unable to trigger a new build for %s.\n", entry)
-			}
-
-			fmt.Printf("Starting new build %d for %s\n", build.Number, entry)
 
-		} else {
-			// get the latest build for the specified repository
-			build, err := client.BuildLast(owner, name, branch)
-			if err != nil {
-				return fmt.Errorf("Error: unable to get latest build for %s.\n", entry)
+		timeout := time.After(p.Timeout)
+		tick := time.Tick(500 * time.Millisecond)
+	
+		// Keep trying until we're timed out, successful or got an error
+		// Tagged with "I" due to break nested in select
+                I:
+		for {
+			select {
+			// Got a timeout! fail with a timeout error
+			case <-timeout:
+				return fmt.Errorf("Error: timed out waiting on a build for %s.\n", entry)
```

Comment:
I've added some clarification to the output indicating what is being waited on. Just let me know if it remains unclear.